google-cloud-storage/lib/google/cloud/storage/project.rb

```diff
@@ -398,21 +396,27 @@ def create_bucket bucket_name, acl: nil, default_acl: nil,
         # steps in [Service Account Authentication](
         # https://cloud.google.com/storage/docs/authentication#service_accounts).
         #
-        # @see https://cloud.google.com/storage/docs/access-control#Signed-URLs
-        #   Access Control Signed URLs guide
+        # @see https://cloud.google.com/storage/docs/access-control/signed-urls
+        #   Signed URLs guide
+        # @see https://cloud.google.com/storage/docs/access-control/signed-urls#signing-resumable
+        #   Using signed URLs with resumable uploads
         #
-        # @param [String] bucket Name of the bucket.
+        # @param [String, nil] bucket Name of the bucket, or nil if URL for all
+        #   buckets is desired.
         # @param [String] path Path to the file in Google Cloud Storage.
         # @param [String] method The HTTP verb to be used with the signed URL.
         #   Signed URLs can be used
         #   with `GET`, `HEAD`, `PUT`, and `DELETE` requests. Default is `GET`.
         # @param [Integer] expires The number of seconds until the URL expires.
-        #   Default is 300/5 minutes.
+        #   If the `version` is `:v2`, the default is 300 (5 minutes). If the
+        #   `version` is `:v4`, the default is 604800 (7 days).
         # @param [String] content_type When provided, the client (browser) must
-        #   send this value in the HTTP header. e.g. `text/plain`
+        #   send this value in the HTTP header. e.g. `text/plain`. This param is
```

Comment:
What if the method added a `Content-Type` header when this argument is provided when using `v4`?

Reply:
I like that idea. So for the `content_type` param, if the user passes it, and there is already `content-type` entry in the `headers` param, what would you do?

Reply:
The description of the `headers` param is:

> Google extension headers (custom HTTP headers that begin with `x-goog-`) that must be included in requests that use the signed URL.

Because it specifies the headers as only "Google extension headers", I don't think we should add a `Content-Type` header. I think we should leave this as-is. The [acceptance tests](https://github.com/googleapis/google-cloud-ruby/blob/9537343f7d4796c0c1e1a6d3976c3a9711f88893/google-cloud-storage/acceptance/storage/signed_url_test.rb#L97-L135) demonstrate that `Content-Type` is not needed for V4 POST URLs to succeed.

@frankyn Sound OK?

Reply:
Gotcha. I'm satisfied.

Reply:
+1 sounds good to me. Completely missed this comment.